Add mark as not done

EswarBalasubramanian opened the pull request.

bittu1040 commented:
Thanks for the PR, 
Can we any how manage both done and not done in single api by using flag ?

EswarBalasubramanian commented:
> Thanks for the PR, Can we any how manage both done and not done in single api by using flag ?

```
router.patch("/done/:id", authMiddleware, async (req, res) => {
  try {
    const { done } = req.body;

    if (done === undefined) {
      return res.status(400).json({ message: "Missing 'done' field in request body" });
    }

    if (typeof done !== "boolean") {
      return res.status(400).json({ message: "'done must be a boolean (true or false)" });
    }

    const task = await Task.findById(req.params.id);

    if (!task) return res.status(404).json({ message: "Task not found" });

    if (task.user.toString() !== req.user.id)
      return res.status(401).json({ message: "Not authorized" });

    task.done = done;
    await task.save();

    res.json({ message: `Task marked as ${done ? "done" : "not done"}`, task });
  } catch (error) {
    console.error("Update Task Status Error:", error);
    res.status(500).json({ message: "Server error", error: error.message });
  }
});
```

- Combined "Mark as Done" and "Mark as Not Done" routes into a single endpoint.
- Added done field in the request body to determine task status.
- Implemented validation to ensure done is a boolean.
- Updated response message to reflect the updated task status.

Let me know if you have any suggestions for improvements.
